app/services/promo_registrations_stats_fetcher.rb

```diff
@@ -10,20 +10,24 @@ def initialize(promo_registrations:)
 
   def perform
     return perform_offline if Rails.application.secrets[:api_promo_base_uri].blank?
-    response = connection.get do |request|
-      request.options.params_encoder = Faraday::FlatParamsEncoder
-      request.headers["Authorization"] = api_authorization_header
-      request.headers["Content-Type"] = "application/json"
-      request.url("/api/2/promo/statsByReferralCode#{query_string}")
-    end
-    referral_code_events_date_list = JSON.parse(response.body)
-
-    @referral_codes.each do |referral_code|
-      promo_registration = PromoRegistration.find_by_referral_code(referral_code)
-      promo_registration.stats = referral_code_events_date_list.select {|referral_code_event_date|
-        referral_code_event_date["referral_code"] == referral_code
-      }.to_json
-      promo_registration.save!
```

Comment:
One thing to note is that this method no longer returns true or false and instead returns nil
That will cause this job to obtain the incorrect result `app/jobs/sync/unattached_promo_registrations_stats_job.rb`

Reply:
hm actually this will return`@referral_codes.each_slice(1000).to_a` if successful, so `app/jobs/sync/unattached_promo_registrations_stats_job.rb` should be fine